components/dashboard/src/settings/ProfileInformation.tsx

```diff
@@ -0,0 +1,202 @@
+        if (state.email.trim() === "") {
+            return "Email must not be empty.";
+        }
+        if (!/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(state.email.trim())) {
+            return "Please enter a valid email.";
+        }
+        return "";
+    }
+
+    export function NudgeForProfileUpdateModal() {
+        const { user, setUser } = useContext(UserContext);
+        const original = ProfileState.getProfileState(user!);
+        const [profileState, setProfileState] = useState(original);
+        const [errorMessage, setErrorMessage] = useState("");
+        const [visible, setVisible] = useState(shouldNudgeForUpdate(user!));
+
+        const saveProfileState = () => {
+            const error = ProfileState.validate(profileState);
+            setErrorMessage(error);
+            if (error) {
+                return;
+            }
+            const updatedUser = ProfileState.setProfileState(user!, profileState);
+            setUser(updatedUser);
+            getGitpodService().server.updateLoggedInUser(updatedUser);
+            setVisible(shouldNudgeForUpdate(updatedUser!));
+        };
+
+        const cancelProfileUpdate = () => {
+            setProfileState(original);
+            saveProfileState();
+        };
+
+        return (
+            <Modal
+                title="Update Your Profile Information"
+                visible={visible}
+                onClose={cancelProfileUpdate}
+                closeable={true}
+                className="_max-w-xl"
+                buttons={
+                    <div>
+                        <button className="secondary" onClick={cancelProfileUpdate}>
+                            Cancel
+                        </button>
+                        <button onClick={saveProfileState}>Save</button>
+                    </div>
+                }
+            >
+                <ProfileInformation
+                    profileState={profileState}
+                    errorMessage={errorMessage}
+                    updated={false}
+                    setProfileState={setProfileState}
+                />
+            </Modal>
+        );
+    }
+}
+
+export default function ProfileInformation(props: {
+    profileState: ProfileState.ProfileState;
+    setProfileState: (newState: ProfileState.ProfileState) => void;
+    errorMessage: string;
+    updated: boolean;
+    children?: React.ReactChild[] | React.ReactChild;
+}) {
+    return (
+        <div>
+            <h3>Profile</h3>
+            <p className="text-base text-gray-500 pb-4 max-w-2xl">
+                The following information will be used to set up Git configuration. You can override Git author name and
+                email per project by using the default environment variables <CodeText>GIT_AUTHOR_NAME</CodeText>,{" "}
+                <CodeText>GIT_COMMITTER_NAME</CodeText>, <CodeText>GIT_AUTHOR_EMAIL</CodeText> and{" "}
+                <CodeText>GIT_COMMITTER_EMAIL</CodeText>.
+            </p>
```

Comment:
**issue**: We probably don't need this part in the modal, right? We could simplify the paragraph to resolve this issue.

Reply:
I think we should keep it. It's actually great for users to get reminded with what name and email they are going to commit when using Gitpod.

Reply:
DEAL—Let's keep it for now. 🤝